src/renderers/shared/reconciler/ReactMultiChild.js

```diff
@@ -342,7 +344,7 @@ var ReactMultiChild = {
           );
         }
         nextIndex++;
-        lastPlacedNode = ReactReconciler.getNativeNode(nextChild);
+        lastPlacedNode = ReactReconciler.getLastNativeNode(nextChild);
```

Comment:
Perhaps getNativeNode can return some other object, perhaps with the pair of opening and closing nodes? You shouldn't need to change the API of ReactReconciler or ReactMultiChild, I think. It doesn't need to be the actual DOM node. The only things that use the output of getNativeNode are DOMChildrenOperations and replaceNodeWithMarkup I think.

(Though it would be nice if getNativeNode doesn't have to allocate because it's called on every reconciliation. I was thinking that perhaps we can just return the closing or opening comment here and traverse to find the other one when we need it but if you really do need both, we should at least cache whatever object to be returned on the text component instance so that we don't allocate each time getNativeNode is called.)